From a332fb796fbe002282d2eb52eee429f7afd03929 Mon Sep 17 00:00:00 2001
From: Vaibhav Bedia <vaibhav.bedia@ti.com>
Date: Tue, 17 Apr 2012 15:05:01 +0530
Subject: [PATCH 0932/1092] ARM: OMAP: AM33XX: Update the EMIF parameters

EMIF parameters are calculated based on the AC timing
parameters from the SDRAM datasheet and the DDR frequency.

Current values for these paramters in AM335x U-Boot code,
though reliable, are not fully optimal. The most optimal
settings can be derived based on the guidelines published
at [1]. A pre-computed set of values with the most optimum
settings for AM335x EVM and BeagleBone can be found at [2].

[1] http://processors.wiki.ti.com/index.php/AM335x_EMIF_Configuration_tips
[2] http://processors.wiki.ti.com/index.php/OMAP_and_Sitara_CCS_support#AM335x

Signed-off-by: Vaibhav Bedia <vaibhav.bedia@ti.com>
---
 arch/arm/mach-omap2/pm33xx.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/arm/mach-omap2/pm33xx.h b/arch/arm/mach-omap2/pm33xx.h
index 3a00eaa..f72c28e 100644
--- a/arch/arm/mach-omap2/pm33xx.h
+++ b/arch/arm/mach-omap2/pm33xx.h
@@ -114,11 +114,11 @@ struct am33xx_padconf {
 #define DATA1_RANK0_DELAYS_0		(DDR_PHY_BASE_ADDR + 0x1D8)
 
 /* Temp placeholder for the values we want in the registers */
-#define EMIF_READ_LATENCY	0x04
-#define EMIF_TIM1		0x0666B3D6
-#define EMIF_TIM2		0x143731DA
-#define EMIF_TIM3		0x00000347
-#define EMIF_SDCFG		0x43805332
+#define EMIF_READ_LATENCY	0x100005	/* Enable Dynamic Power Down */
+#define EMIF_TIM1		0x0666B3C9
+#define EMIF_TIM2		0x243631CA
+#define EMIF_TIM3		0x0000033F
+#define EMIF_SDCFG		0x41805332
 #define EMIF_SDREF		0x0000081a
 #define EMIF_SDMGT		0x80000000
 #define EMIF_SDRAM		0x00004650
-- 
1.7.11.2

